source/base/configbase.h

```diff
@@ -680,6 +680,17 @@
     #undef POV_SYS_IMAGE_EXTENSION
 #endif
 
+/// @def POV_FILENAME_BUFFER_CHARS
+/// The number of characters to reserve for file name buffers.
+///
+/// This setting is used in allocating temporary buffers to construct file names, and should be set
+/// to the maximum number of ASCII characters in a file name the system can handle safely.
+/// The value is understood to exclude any terminating NUL character.
+///
+#ifndef POV_FILENAME_BUFFER_CHARS
+    #define POV_FILENAME_BUFFER_CHARS 199
+#endif
+
```

Comment:
The newly introduced system-specific compile-time configuration setting `POV_FILENAME_BUFFER_CHARS` replaces `FILE_NAME_LENGTH`, to achieve the following goals:
- Clarify whether the size does or does not include room for a terminating NUL character. (The old macro de-facto did, but this was not documented anywhere, and the name might have been interpreted otherwise; the new macro is clearly documented, and the name chosen accordingly, to exclude any terminating NUL character. Net size was favoured over gross size to better fit with C++-style strings.)
- Clarify whether the size should reflect the system's _absolute_ limit for file name size, the _safe_ limit, or something else entirely. (The new macro is now specified to be just the size for internal buffers in POV-Ray; using the maximum safe limit is just a recommendation.)
- Clarify how the macro's value should be interpreted in wide or multi-byte character contexts. (With the new macro no longer strictly tied to system file name limits, this should now be less of an issue.)